cmd/flipt/export.go

```diff
@@ -51,27 +133,59 @@ func runExport(ctx context.Context, logger *zap.Logger) error {
 		store = mysql.NewStore(db, logger)
 	}
 
-	// default to stdout
-	var out io.WriteCloser = os.Stdout
+	migrator, err := sql.NewMigrator(*cfg, logger)
+	if err != nil {
+		return nil, nil, err
+	}
 
-	// export to file
-	if exportFilename != "" {
-		logger.Debug("exporting", zap.String("destination_path", exportFilename))
+	defer migrator.Close()
 
-		out, err = os.Create(exportFilename)
-		if err != nil {
-			return fmt.Errorf("creating output file: %w", err)
+	// drop tables if specified
+	if dropBeforeMigrate {
+		logger.Debug("dropping tables")
+
+		if err := migrator.Down(); err != nil {
+			return nil, nil, fmt.Errorf("attempting to drop: %w", err)
 		}
+	}
 
-		fmt.Fprintf(out, "# exported by Flipt (%s) on %s\n\n", version, time.Now().UTC().Format(time.RFC3339))
+	if err := migrator.Up(forceMigrate); err != nil {
+		return nil, nil, err
 	}
 
-	defer out.Close()
+	if _, err := migrator.Close(); err != nil {
+		return nil, nil, fmt.Errorf("closing migrator: %w", err)
+	}
+
+	return server.New(logger, store), func() { _ = db.Close() }, nil
+}
+
+func fliptClient(logger *zap.Logger, address, token string) *sdk.Flipt {
+	addr, err := url.Parse(address)
+	if err != nil {
+		logger.Fatal("Export address is invalid", zap.Error(err))
+	}
+
+	var transport sdk.Transport
+	switch addr.Scheme {
+	case "http":
+		transport = sdkhttp.NewTransport(address)
+	case "grpc":
+		conn, err := grpc.Dial(addr.Host,
+			grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			logger.Fatal("Failed to dial Flipt", zap.Error(err))
+		}
+
+		transport = sdkgrpc.NewTransport(conn)
+	}
```

Comment:
Should we guard against the case of a valid but unsupported scheme?

Reply:
💯 good spot.